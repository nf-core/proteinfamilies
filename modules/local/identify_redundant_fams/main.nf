process IDENTIFY_REDUNDANT_FAMS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/4a/4af7c28266e09eb3af86e75b1c61e9876d2827acddfe793cef5a704a1e98fb9b/data' :
        'community.wave.seqera.io/library/pandas_python:f5ae1ede7f6e9d73' }"

    input:
    tuple val(meta) , path(mapping)
    tuple val(meta2), path(domtbl)
    val(length_threshold)

    output:
    tuple val(meta), path("redundant_fam_ids.txt"), emit: redundant_ids
    path "versions.yml"                           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    """
    identify_redundant_fams.py \\
        --mapping ${mapping} \\
        --domtbl ${domtbl} \\
        --length_threshold ${length_threshold} \\
        --out_file redundant_fam_ids.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        pandas: \$(python -c "import importlib.metadata; print(importlib.metadata.version('pandas'))")
    END_VERSIONS
    """

    stub:
    """
    touch redundant_fam_ids.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        pandas: \$(python -c "import importlib.metadata; print(importlib.metadata.version('pandas'))")
    END_VERSIONS
    """
}
