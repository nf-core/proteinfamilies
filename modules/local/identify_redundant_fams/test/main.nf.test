nextflow_process {

    name "Test Process IDENTIFY_REDUNDANT_FAMS"
    script "../main.nf"
    process "IDENTIFY_REDUNDANT_FAMS"

    test("proteinfamilies - family redundancy") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'mqc' ],
                    file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_meta_mqc.csv', checkIfExists: true),
                ])
                input[1] = Channel.of([
                    [ id: 'domtbl' ],
                    file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test.domtbl.gz', checkIfExists: true),
                ])
                input[2] = 0.9
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.redundant_ids,
                    process.out.versions.collect { path(it).yaml }
                    ).match()}
            )
        }
    }

    test("proteinfamilies - family redundancy - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'mqc' ],
                    file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_meta_mqc.csv', checkIfExists: true),
                ])
                input[1] = Channel.of([
                    [ id: 'domtbl' ],
                    file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test.domtbl.gz', checkIfExists: true),
                ])
                input[2] = 0.9
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
