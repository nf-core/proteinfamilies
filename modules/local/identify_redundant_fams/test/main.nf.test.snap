{
    "proteinfamilies - family redundancy - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mqc"
                        },
                        "redundant_fam_ids.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ee9740f51b2073a4fc34f9ba4b2d0022"
                ],
                "redundant_ids": [
                    [
                        {
                            "id": "mqc"
                        },
                        "redundant_fam_ids.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ee9740f51b2073a4fc34f9ba4b2d0022"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-13T15:29:51.689666568"
    },
    "proteinfamilies - family redundancy": {
        "content": [
            [
                [
                    {
                        "id": "mqc"
                    },
                    "redundant_fam_ids.txt:md5,d54a9f9187d78a1566ec3156b96f2f78"
                ]
            ],
            [
                {
                    "IDENTIFY_REDUNDANT_FAMS": {
                        "python": "3.13.1",
                        "pandas": "2.3.0"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-06-13T15:29:44.618846903"
    }
}